### PR TITLE
More xml docs

### PR DESCRIPTION
**Pull request by danmoseley**
Contributes to https://github.com/dotnet/runtime/issues/54871 .. after this I can port the M.W32 changes into docs.

I also ran the reverse port tool for System.IO types and figured i might as well commit the changes that were easy/obviously correct. To make System.IO the source of truth we'd need to run it again and address issues in the stuff I didn't include here.

### CODE REVIEW

File: src/libraries/System.Diagnostics.Process/src/Microsoft/Win32/SafeHandles/SafeProcessHandle.cs [hunk 1]
```diff
@@ -19,6 +19,9 @@ public sealed partial class SafeProcessHandle : SafeHandleZeroOrMinusOneIsInvali
     {
         internal static readonly SafeProcessHandle InvalidHandle = new SafeProcessHandle();
 
+        /// <summary>
+        /// Creates a SafeProcessHandle
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/src/Microsoft/Win32/SafeHandles/SafeProcessHandle.cs:23]:
```suggestion
        /// Creates a SafeProcessHandle.
```

Comment (stephentoub) [reply src/libraries/System.Diagnostics.Process/src/Microsoft/Win32/SafeHandles/SafeProcessHandle.cs:23]: Also, should the `SafeProcessHandle` be `<see cref="SafeProcessHandle"/>`?  Same question for other occurrences.

File: src/libraries/System.Diagnostics.Process/src/Microsoft/Win32/SafeHandles/SafeProcessHandle.cs [hunk 2]
```diff
@@ -29,6 +32,11 @@ internal SafeProcessHandle(IntPtr handle)
         {
         }
 
+        /// <summary>
+        /// Creates a SafeProcessHandle around a Windows process handle.
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/src/Microsoft/Win32/SafeHandles/SafeProcessHandle.cs:36]: What about on other operating systems?

File: src/libraries/System.IO.MemoryMappedFiles/src/Microsoft/Win32/SafeMemoryMappedFileHandle.cs [hunk 3]
```diff
@@ -7,6 +7,9 @@ namespace Microsoft.Win32.SafeHandles
 {
     public sealed partial class SafeMemoryMappedFileHandle : SafeHandleZeroOrMinusOneIsInvalid
     {
+        /// <summary>
+        /// Creates a SafeMemoryMappedFileHandle
```
Comment (stephentoub) [inline src/libraries/System.IO.MemoryMappedFiles/src/Microsoft/Win32/SafeMemoryMappedFileHandle.cs:11]:
```suggestion
        /// Creates a SafeMemoryMappedFileHandle.
```

File: src/libraries/System.IO.Pipes/src/Microsoft/Win32/SafeHandles/SafePipeHandle.cs [hunk 4]
```diff
@@ -14,6 +14,11 @@ public SafePipeHandle()
         {
         }
 
+        /// <summary>
+        /// Creates a SafePipeHandle around a Windows pipe handle.
```
Comment (stephentoub) [inline src/libraries/System.IO.Pipes/src/Microsoft/Win32/SafeHandles/SafePipeHandle.cs:18]: Same OS question.

File: src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs [hunk 5]
```diff
@@ -75,6 +77,81 @@ private void Init()
             }
         }
 
+        /// <returns>'true' if new data was found</returns>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs:80]: Strange to see an XML comment that includes "returns" but not "summary".

**Comment by danmoseley** [comment]
updated

**Comment by danmoseley** [comment]
The other failure was the xunit crash. @stephentoub any other feedback?

**Comment by MattGal** [comment]
> dotnet/dnceng there seems to be a problem gathering results from one pipeline. log ends with
> 
> ```
> INFO: pip is looking at multiple versions of azure-devops to determine which version is compatible with other requirements. This could take a while.
> ERROR: Cannot install msrest because these package versions have conflicting dependencies.
> 
> The conflict is caused by:
>     requests 2.25.1 depends on chardet<5 and >=3.0.2
>     requests 2.25.0 depends on chardet<4 and >=3.0.2
>     requests 2.24.0 depends on chardet<4 and >=3.0.2
>     requests 2.23.0 depends on chardet<4 and >=3.0.2
>     requests 2.22.0 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.21.0 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.20.1 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.20.0 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.19.1 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.19.0 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.18.4 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.18.3 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.18.2 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.18.1 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.18.0 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.17.3 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.17.2 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.17.1 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.17.0 depends on chardet<3.1.0 and >=3.0.2
> ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
>     requests 2.16.5 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.16.4 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.16.3 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.16.2 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.16.1 depends on chardet<3.1.0 and >=3.0.2
>     requests 2.16.0 depends on chardet<3.1.0 and >=3.0.2
> 
> To fix this you could try to:
> 1. loosen the range of package versions you've specified
> 2. remove package versions to allow pip attempt to solve the dependency conflict
> 
> + /home/helixbot/.azdo-env/bin/python -c import future
> ```
> 
> although it seems the tests passed fine: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-57039-merge-573ed31f60a142e5a6/Microsoft.Extensions.Hosting.Unit.Tests/1/console.5f8543fb.log?sv=2019-07-07&se=2021-09-01T04%3A10%3A16Z&sr=c&sp=rl&sig=pvKU9JNkyySgmxIGYSDrEfqlK5TY5HPu9NhFsku6bLE%3D

This reporter is scheduled to be EOL very soon via @ChadNedzlek 's very cool work putting the reporting into the Helix client itself.  That said, it's weird this wouldn't be a 100% failure so I'll spend some time looking at why just in case there's a cheap mitigation

**Comment by MattGal** [comment]
@danmoseley while the logging is weird, given 246 other work items did the same thing successfully this looks like it was a transient problem with pypi.org.  The Helix client will have this preinstalled in the future which will both save a ton of time (especially in docker scenarios since it can't cache anything between work items) and prevent this sort of weirdness.

File: src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs [hunk 6]
```diff
@@ -75,6 +77,84 @@ private void Init()
             }
         }
 
+        /// <summary>
+        /// Fills the buffer with the next set of data.
+        /// </summary>
+        /// <returns>'true' if new data was found</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe bool GetData()
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs:85]: I assume all this code is exactly as it was in the other file?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs (hunk 6 above)
Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs:85]: Yes

**Comment by danmoseley** [comment]
I assume this a symptom of the issue we have ingested SDK for
```
❌[Log] .dotnet/sdk/6.0.100-preview.6.21355.2/NuGet.RestoreEx.targets(19,5): error : Failed to retrieve information about 'System.IO.Ports' from remote source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/\_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/system.io.ports/index.json'.
at NuGet.Protocol.HttpFileSystemBasedFindPackageByIdResource.FindPackagesByIdAsync(String id, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken) in NuGet.Protocol.dll:token 0x6000393+0x2dc
```

**Comment by MattGal** [comment]
> I assume this a symptom of the issue we have ingested SDK for
> 
> ```
> ❌[Log] .dotnet/sdk/6.0.100-preview.6.21355.2/NuGet.RestoreEx.targets(19,5): error : Failed to retrieve information about 'System.IO.Ports' from remote source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/\_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/system.io.ports/index.json'.
> at NuGet.Protocol.HttpFileSystemBasedFindPackageByIdResource.FindPackagesByIdAsync(String id, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken) in NuGet.Protocol.dll:token 0x6000393+0x2dc
> ```

Yup, anywhere you see older SDK and these lines:
```
  An error occurred while sending the request.
    Unable to read data from the transport connection: Connection reset by peer.
    Connection reset by peer
```
... that's how we started to suspect the SDK in the first place, as none of the requests that look like this had any telemetry in Azure Front Door or Azure Package feeds.

**Pull request merged**